docs/blog/rendering-images/index.qmd

```diff
@@ -0,0 +1,292 @@
+```
+1. These image files follow a patterned naming convention, such as `metro_1.svg`, `metro_2.svg`, and so on.
+
+
+Our goal is to construct a Python dictionary to serve as input for creating a `Pandas` DataFrame.
+This dictionary will include the previously defined `data` and a new key, `"case1"`. The value for
+`"case1"` is generated by processing the `data["lines"]` column, which contains a list of strings.
+
+For each string in `data["lines"]`, we split it by commas to extract individual numbers. These
+numbers are then processed by adding the necessary prefix and suffix to generate valid file paths.
+Once processed, the paths are joined back into a single string using `", ".join()`. The resulting
+strings from all iterations are collected into a list, which becomes the value for `"case1"`:
+```{python}
+# | code-fold: true
+metro_mini1 = pd.DataFrame(
+    {
+        **data,
+        "case1": [
+            ", ".join(
+                str((img_local_paths / f"metro_{item}").with_suffix(".svg"))
+                for item in row.split(", ")
+            )
+            for row in data["lines"]
+        ],
+    }
+)
+metro_mini1
+```
+
+::: {.callout-tip collapse="false"}
+## Use the `Pathlib` Module to Construct Paths
+
+Local file paths can vary depending on the operating system, which makes it easy to accidentally
+construct invalid paths. A good practice to mitigate this is to use Python's built-in
+[Pathlib](https://docs.python.org/3/library/pathlib.html) module to construct paths first and then
+convert them to strings.
+:::
+
+Now that the `case1` column of `metro_mini1` contains the correct local file paths, we can use
+`GT.fmt_image()` to render the images by providing `"case1"` as the first argument:
+```{python}
+GT(metro_mini1).fmt_image("case1")
+```
+
+### Case 2: Full HTTP/HTTPS URLs
+**Case 2** demonstrates how to simulate a column containing strings representing HTTP/HTTPS URLs. We'll
+use the same images as in **Case 1**, but this time, retrieve them from the Great Tables GitHub repository:
+```{python}
+img_url_paths = "https://raw.githubusercontent.com/posit-dev/great-tables/refs/heads/main/great_tables/data/metro_images"
+```
+
+Preparing a Python dictionary as input for creating a `Pandas` DataFrame follows a similar process
+to **Case 1**. However, in this case, the new key is named `case2`:
+```{python}
+# | code-fold: true
+metro_mini2 = pd.DataFrame(
+    {
+        **data,
+        "case2": [
+            ", ".join(f"{img_url_paths}/metro_{item}.svg" for item in row.split(", "))
+            for row in data["lines"]
+        ],
+    }
+)
+metro_mini2
+```
+
+In the same way, we can use `GT.fmt_image()` to render the images by passing `"case2"` as the first
+argument:
+```{python}
+GT(metro_mini2).fmt_image("case2")
+```
+
+
+### Case 3: File Names with the `path=` Argument
+**Case 3** demonstrates how to simulate a column containing file names. The preparation for creating
```

Comment:
A sentence on why might help prime readers for what they'll get out of this case. Maybe something like...

> **Case 3** demonstrates using the `path=` argument to specify images relative to a base directory. This lets you cut out a lot of repetition in file names. \<remove DataFrame code prep explanation text\>